src/codegen/enums.rs

```diff
@@ -150,6 +151,25 @@ fn generate_enum(env: &Env, w: &mut Write, enum_: &Enumeration, config: &GObject
 "
     ));
 
+    // Generate Display trait implementation.
+    try!(cfg_deprecated(w, env, enum_.deprecated_version, false, 0));
+    try!(version_condition(w, env, enum_.version, false, 0));
+    try!(writeln!(w,
+                  "impl fmt::Display for {0} {{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {{
+        write!(f, \"{0}::{{}}\", match *self {{", enum_.name));
+    for member in &members {
+        try!(cfg_deprecated(w, env, member.deprecated_version, false, 3));
+        try!(version_condition(w, env, member.version, false, 3));
+        try!(writeln!(w, "\t\t\t{0}::{1} => \"{1}\",", enum_.name, member.name));
+    }
+    try!(writeln!(w,
+                  "\t\t\t_ => \"Unknown\",
+        }})
+    }}
+}}"));
```

Comment:
IMHO extra new line needed here before next `#[doc(hidden)] impl ToGlib for xx {`